gateway/redis_analytics_purger.go

```diff
@@ -34,12 +34,17 @@ func (r RedisPurger) PurgeLoop(ctx context.Context) {
 
 func (r *RedisPurger) PurgeCache() {
 	expireAfter := config.Global().AnalyticsConfig.StorageExpirationTime
+
+	if expireAfter == -1 {
+		return
+	}
+
 	if expireAfter == 0 {
 		expireAfter = 60 // 1 minute
 	}
 
 	exp, _ := r.Store.GetExp(analyticsKeyName)
-	if exp <= 0 {
+	if exp <= -1 {
```

Comment:
@sredxny this should be exp = -1. If TTL is -2, it means that the key doesn't exist so we don't want to set anything there.